File: numbers/sum_of_numbers/sum_of_numbers.rb
def sum_of_numbers(number)
  sum = 0
  for number in 0..number do
    sum = number + sum
    number += 1
  end
  return sum
end

puts "Enter a number: "
number = gets.to_i
result = sum_of_numbers(number)

puts "\nThe sum of numbers is: #{result}"

File: numbers/sum_of_numbers/sum_of_numbers.js
// Sum of numbers

const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

function sumOfNumbersIterative(n) {
  let totalSum = 0;
  for (let i = 0; i < n; i++) {
    totalSum = totalSum + (i + 1);
  }
  return totalSum;
}

readline.question("Enter a number: ", (number) => {
  console.log(sumOfNumbersIterative(number));
  readline.close();
});
